### PR TITLE
Implement save,delete,edit,share TT and delete section feature 

### PR DESCRIPTION
**Pull request by Aviral14**
- Create `/api/timetable/save` `/api/timetable/getTT` `/api/timetable/delete` `/api/timetable/toggleShare` `/api/timetable/viewShared/` api routes to save,fetch,delete,shared and retrieve shared TimeTables respectively.
- Modify `Dashboard` Component to display saved TimeTables and delete and share them.
-  Add  Save Timetable button to `CreateTimeTable` component.
- Implement delete section on clicking TimeTable tile functionality.
- Add PropTypes to components.

### CODE REVIEW

**Comment by AetherPrior** [comment]
This also fixes the onClick problem

**Review by AetherPrior — APPROVED** [review]
Have seen the code before the PR.

File: models/schemas/Entry.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  constructor(code, name, room, len, section) {
+    this.courseCode = code;
+    this.courseName = name;
+    this.sectionRoom = room;
+    this.numHours = len;
+    this.section = section;
+  }
+}
+
+module.export = Entry;
```
Comment (VikramjeetD) [inline models/schemas/Entry.js:11]: Why are these redeclared? Is it not possible to access what is inside `/client` from here?

Comment (Aviral14) [reply models/schemas/Entry.js:11]: yes

**Pull request merged**